pkgs/development/tools/protoc-gen-go-grpc/default.nix

```diff
@@ -0,0 +1,29 @@
+{ buildGoPackage
+, fetchFromGitHub
+, lib
+}:
+
+buildGoPackage rec {
```

Comment:
Please use buildGoModule.

Reply:
As mentioned in the PR description and the commit message, 

> Using `buildGoModule` did not work for this package, and produced the error "main module (google.golang.org/grpc) does not contain package google.golang.org/grpc/cmd/protoc-gen-go-grpc". Since `buildGoModule` and `buildGoPackage` both seem in common use and both have their advantages (#84826) just using `buildGoPackage` seems like a reasonable solution here.

Do you disagree? Why?

Reply:
With `buildGoModule` you just need use `modRoot` argument instead of `subPackages`. It is not subpackage but submodule.

Reply:
That indeed works, thanks!

Do you have any insight into whether we should prefer `buildGoModule` or `buildGoPackage`? #84826 seemed somewhat inconclusive (to my untrained eyes)

Reply:
No, not really. However for me `buildGoModule` is easier for use with go modules and will probaly support go 1.17 out of the box.